infra/src/main/kotlin/fi/fta/geoviite/infra/split/SplitDao.kt

```diff
@@ -105,42 +124,26 @@ class SplitDao(
               split.bulk_transfer_state,
               split.publication_id,
               split.source_location_track_id,
-              array_agg(split_relinked_switch.switch_id) as switch_ids
+              array_agg(split_relinked_switch.switch_id) as switch_ids,
+              array_agg(split_updated_duplicate.duplicate_id) as updated_duplicate_ids
           from publication.split 
               left join publication.split_relinked_switch on split.id = split_relinked_switch.split_id
+              left join publication.split_updated_duplicate on split.id = split_updated_duplicate.split_id
           where id = :id
           group by split.id
         """.trimIndent()
 
     fun get(splitId: IntId<Split>): Split? {
         return jdbcTemplate.queryOptional(splitFetchSql, mapOf("id" to splitId.intValue)) { rs, _ ->
-            val targetLocationTracks = getSplitTargets(splitId)
-
-            Split(
-                id = splitId,
-                locationTrackId = rs.getIntId("source_location_track_id"),
-                bulkTransferState = rs.getEnum("bulk_transfer_state"),
-                publicationId = rs.getIntIdOrNull("publication_id"),
-                targetLocationTracks = targetLocationTracks,
-                relinkedSwitches = rs.getIntIdArray("switch_ids"),
-            )
+            splitFromResultSet(rs)
```

Comment:
Näiden Split()-dataluokkien luonti näytti hieman toisteiselta, omasta mielestä vaikutti järkevältä funktioida.

Monissa ohjelmointikielissä tällaiset JokinDataLuokka.from(jokinToinenHärpäke) on muuten aika yleisiä. Periaatteessa tällaisen voisi ihan vastaavasti luoda companion objektiin (ns. Split.from(resultSet)), mutta sitten siellä olisi taas tarpeellista tietää kantasarakkeiden nimet joka olisi rakenteen puolesta ikävämpää (nyt SQL-mälväys on daossa)

Mielipiteitä? Mietin myös jotain createSplitFromResult()-nimitystä, mutta se vaikuttaisi että resultSettiä jotenkin splitattaisiin (😄).

Reply:
Selkein paikka varmaan ResultSetExternalissa extension methodina, siellähän meillä on noita muitakin olioiden lukemisia ResultSeteistä vaikka kuinka paljon.

Reply:
Totta! Sinänsä se voisikin olla sopiva paikka, joskin rakenteena se vaikuttaisi kyllä vähän skaalautumattomalta 🤔 (jos kaikkien tyyppien luonnit ResultSeteistä laitettaisiin aina tuonne, niin filun koko kasvaisi aika nopeaa tahtia). Samalla sinne päätyisi aika monesta eri SQL-taulusta määritelmätietoa, joka varmaan haluttaisiin pitää rajatussa (samassa) paikassa löydettävyyden ylläpitämiseksi.

Suurinhan osa tuon ResultSetExternalin sisällöstä näyttäisi nykytilanteessa olevan melko geneeristä util-funkkaria, ja moni niistäkin vaikuttaisi olevan useammassa kuin yhdessä paikassa/DAOssa käytössä, joka tuntuukin intuitiiviselta rakenteelta. Sen sijaan aika harvasta eri paikasta/DAOsta tarvitsisi varmaankaan Split-dataluokkien instansseja kuitenkaan luoda ResultSeteistä, jolloin tällaisen luontifunkkarin sijoittaminen sinne vaikuttaisi hieman oudolta ratkaisulta.

Sen sijaan tuollaisen extension metodin voisi tietysti laittaa aluksi samaan tiedostoon kuin tuo DAO, jolloin sen voisi tarvittaessa siirtää useamman paikan tarvitsemana tuonne ResultSetExternaliin, joka vaikuttaisi ainakin minusta ihan mukavalta rakenteelta (taidan vaihtaakin tällaiseksi ilman parempia ideoita tai suuntaviittoja).

Toki näitä voisi vielä ylikuormittaa eri parametreillakin tarvittaessa (riippuen mitä kysely palauttaa/mitä ennakkotietoja oli ennen kyselyä), joka myös vaikuttaisi ihan mielekkäältä.

```
ResultSet.toSplit(...) = Split(
    ...
)